linux/centos.dockerfile

```diff
@@ -1,12 +1,20 @@
 FROM centos:centos8 as centos-base
 
+ARG STAGING_REPO=false
+
 # Installing needed tools
 RUN yum update -y && \
     yum install -y wget
 
 # Adding Newrelic repository
-RUN wget -O /etc/yum.repos.d/newrelic-infra.repo https://download.newrelic.com/infrastructure_agent/linux/yum/el/8/x86_64/newrelic-infra.repo && \
-    yum -q makecache -y --disablerepo='*' --enablerepo='newrelic-infra' && \
+RUN if [ "${STAGING_REPO}" = "true" ]; then \
+        repo="http://nr-downloads-ohai-staging.s3-website-us-east-1.amazonaws.com/infrastructure_agent/linux/yum/el/8/x86_64/newrelic-infra.repo"; \
+    else \
+        repo="https://download.newrelic.com/infrastructure_agent/linux/yum/el/8/x86_64/newrelic-infra.repo"; \
+    fi; \
+    wget -O /etc/yum.repos.d/newrelic-infra.repo "$repo"
```

Comment:
perhaps we could parametrize the os version (8) and the arch (x86_64), so we can test different combinations in the future (for instance 9 and arm64). Same for the other OSs.

Reply:
This is now more parametrizable, by adding more distros as `helper_whatever.sh`.
Implementing per-distro parametrization is also possible now with the helpers, but would be more work. I'd leave it out of scope for this PR.